src/main/java/com/ibm/watson/modelmesh/ModelMesh.java

```diff
@@ -3400,7 +3441,9 @@ protected Object invokeModel(final String modelId, final Method method, final Me
                 if (!instanceId.equals(destId)) {
                     if (directClient != null) {
                         logger.info("Forwarding mis-directed request for model " + modelId + " to inst " + destId);
-                        return forwardInvokeModel(destId, modelId, remoteMeth, args);
+                        Object outcome = forwardInvokeModel(destId, modelId, remoteMeth, args);
```

Comment:
should payload processing only happen on the MM instance that actually processed the inference request?

Reply:
that's a very valid point, thanks. I will make sure to execute `PayloadProcessors` only on those instances.